modmesh/app/euler1d.py

```diff
@@ -442,288 +453,266 @@ def _update_single_figure(self):
-        if None is shocktube.svr:
-            raise ValueError("shocktube does not have numerical solver built")
+        self.st.svr.march_alpha2(steps=steps)
+        self.current_step += steps
+        time_current = self.current_step * self.st.svr.time_increment
+        self.st.build_field(t=time_current)
+        cfl = self.st.svr.cfl
+        self.log(f"CFL: min {cfl.min()} max {cfl.max()}")
+        self.update_lines()
+        if self.profiling:
+            self.log(mm.time_registry.report())
 
-        super().__init__()
+    def step(self, steps=1):
+        """
+        Callback function of step button.
 
-        self.shocktube = shocktube
+        :return: None
+        """
+        self.march_alpha2(steps=steps)
 
-        self.max_steps = max_steps
-        self.current_step = 0
-        self.timer = None
-
-        self.use_sub = mm.Toggle.instance.get_value('apps.euler1d.use_sub',
-                                                    False)
-        if self.use_sub is None:
-            self.use_sub = mm.Toggle.instance.solid.use_pyside
-        self._main = QWidget()
-
-        if self.use_sub:
-            # FIXME: sub window causes missing QWindow with the following
-            # error:
-            # RuntimeError:
-            # Internal C++ object (PySide6.QtGui.QWindow) already deleted.
-            # It is probably because RMainWindow is not recognized by PySide6
-            # and matplotlib.  We may consider to use composite for QMainWindow
-            # instead of inheritance.
-            self._subwin = view.mgr.addSubWindow(self._main)
-            self._subwin.resize(1150, 750)
-
-        self.plt_mgr = PlotManager(self.shocktube)
-        self.plt_mgr.build_lines_grid_layout()
-        # Ideally one would use self.addToolBar here, but it is slightly
-        # incompatible between PyQt6 and other bindings, so we just add the
-        # toolbar as a plain widget instead.
-        layout = QVBoxLayout(self._main)
-        ctrl_button_layout = QHBoxLayout()
-
-        start_button = QPushButton()
-        start_button.setText('Start')
-        start_button.clicked.connect(self.start)
-        ctrl_button_layout.addWidget(start_button)
-
-        stop_button = QPushButton()
-        stop_button.setText('Stop')
-        stop_button.clicked.connect(self.stop)
-        ctrl_button_layout.addWidget(stop_button)
-
-        step_button = QPushButton()
-        step_button.setText('Step')
-        step_button.clicked.connect(self.step_button_cb)
-        ctrl_button_layout.addWidget(step_button)
-
-        reset_button = QPushButton()
-        reset_button.setText('Reset')
-        reset_button.clicked.connect(self.reset)
-        ctrl_button_layout.addWidget(reset_button)
-
-        layout.addWidget(self.plt_mgr)
-        layout.addLayout(ctrl_button_layout)
-
-        self.profiling = profiling
-
-    def show(self):
-        self._main.show()
-        if self.use_sub:
-            self._subwin.show()
```

Comment:
Using the `PUI` to instead of these lousy imperative code.
The GUI layout is implemented in [688](https://github.com/solvcon/modmesh/pull/271/files#diff-4ebeb3148c683103bf030637fdf520154a4b12d9b660a82610861aa5250a0be1R688) and [645](https://github.com/solvcon/modmesh/pull/271/files#diff-4ebeb3148c683103bf030637fdf520154a4b12d9b660a82610861aa5250a0be1R645)